api/oneclick.py

```diff
@@ -343,14 +345,35 @@ def update_licensepool_for_identifier(self, isbn, availability):
         if not availability:
             licenses_available = 0
         licenses_owned = licenses_available
+
+
+        # If possible, create a FormatData object representing
+        # how the book is available.
+        formats = []
+
+        # Note that these strings are different from the similar strings
+        # found in "fileFormat" when looking at a patron's loans.
+        # "ebook" (a medium) versus "EPUB" (a format). Unfortunately we
+        # 
+        delivery_type = None
+        medium = medium.lower()
+        if medium == 'ebook':
+            delivery_type = Representation.EPUB_MEDIA_TYPE
+        elif medium == 'eaudio':
+            # A questionable assumption.
+            delivery_type = Representation.MP3_MEDIA_TYPE
+
+        if delivery_type:
+            formats.append(FormatData(delivery_type, None))
```

Comment:
I imported data from a OneClick collection and all the books were missing their delivery mechanisms. This fixes the problem but `isbn_info_to_metadata` (in core/oneclick.py) does create a CirculationData object with the appropriate FormatData, so it shouldn't have been a problem. I wasn't able to investigate why the FormatData isn't created initially. The code in that method like `cls.oneclick_formats.get("audiobook-mp3-oneclick")` seems copied and pasted  from Overdrive code and can be simplified quite a bit.